core/conn/jdbc_type2/native/SrvrCommon.cpp

```diff
@@ -1634,6 +1634,92 @@ short do_ExecSMD(
                         tableParam[0], inputParam[0], inputParam[1],
                         inputParam[2], inputParam[3]);
                     break;
+                case SQL_API_SQLPROCEDURES:
+                    if ((!checkIfWildCard(catalogNm, catalogNmNoEsc) || !checkIfWildCard(schemaNm, schemaNmNoEsc) || !checkIfWildCard(
+tableNm, tableNmNoEsc)) && !metadataId)
+                    {
+                        executeException->exception_nr = odbc_SQLSvc_GetSQLCatalogs_ParamError_exn_;
+                        executeException->u.ParamError.ParamDesc = SQLSVC_EXCEPTION_WILDCARD_NOT_SUPPORTED;
+                        FUNCTION_RETURN_NUMERIC(EXECUTE_EXCEPTION,("EXECUTE_EXCEPTION"));
+                    }
+
+                    if (strcmp(catalogNm,"") == 0)
+                        strcpy(tableName1,SEABASE_MD_CATALOG);
```

Comment:
Should this be changed to the default catalog in the connection when the catalogNm is NULL?  Default to SEABASE instead of SEABASE_MD_CATALOG when it is not set.  Ditto for all other APIs.